python/adbc_driver_manager/adbc_driver_manager/_lib.pyx

```diff
@@ -460,6 +461,18 @@ cdef class _AdbcHandle:
                     f"with open {self._child_type}")
 
 
+cdef void pycapsule_stream_deleter(object stream_capsule):
+    cdef:
+        CArrowArrayStream* stream
+    # Do not invoke the deleter on a used/moved capsule
+    stream = <CArrowArrayStream*>cpython.PyCapsule_GetPointer(
+        stream_capsule, 'arrowarraystream'
+    )
```

Comment:
That is pretty tricky though. PyCapsule_GetPointer will set the global python error but I'm not sure how you'd know to check for that after this is executed; so this could potentially be a pitfall of segfaults or leaked pointers

Reply:
You don't need to check for error, Cython will do it for you thanks to its `PyCapsule_GetPointer` declaration here:
https://github.com/cython/cython/blob/d73164b56544def09b65d250d72b227a38944bb1/Cython/Includes/cpython/pycapsule.pxd#L50

Reply:
As for having a NULL stream pointer in a C ArrowArrayStream capsule, this should probably be disallowed by the spec.